### PR TITLE
Move `Level Control Cluster` to match the spec

### PR DESCRIPTION
**Pull request by andy31415**
## Changes:

- RENAME `MoveMode` to `MoveModeEnum`
- RENAME `StepMode` with `StepModeEnum` (usage of StepMode appearing in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7831)
- RENAME `LevelControlOptions` to `OptionsBitmap`

Along with changes for compilation (zap regen, zap-update-all, adding to backwards compat and replace the usages of the above enums/bitmaps in code where I could find them)

## NOT changed (will change spec)

Spec says OptionsMask is `map8` (untyped) however using OptionsBitmap sems to correct thing to do.

### CODE REVIEW

**Comment by andy31415** [comment]
@bzbarsky-apple this removes StepMode enum and it is supposed to be an alias to MoveModeEnum now. Is this doable in `availability.yaml` ?

**Comment by andy31415** [comment]
> @bzbarsky-apple this removes StepMode enum and it is supposed to be an alias to MoveModeEnum now. Is this doable in `availability.yaml` ?

Added StepModeEnum back because https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7831 adds it in the spec

File: src/app/clusters/level-control/level-control.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/level-control/level-control.cpp:1]: Please consider making this sort of change in a separate PR in the future.  That will:

1. Make the XML PR simpler to review.
2. Exercise the shim names being added.

File: third_party/imgui/repo
Comment (bzbarsky-apple) [inline third_party/imgui/repo:1]: All these third_party changes seem unrelated.  This is the part that's "changes requested".

**Review by bzbarsky-apple — DISMISSED** [review]

File: third_party/imgui/repo
Comment (andy31415) [reply third_party/imgui/repo:1]: Removed those changes.

File: src/app/clusters/level-control/level-control.cpp
Comment (andy31415) [reply src/app/clusters/level-control/level-control.cpp:1]: This balooned larger than expected - will try in the future to make things more reviewable. Also hoping this will not happen as often.

**Review by bzbarsky-apple — APPROVED** [review]
